salt/config/__init__.py

```diff
@@ -3284,6 +3292,42 @@ def _cache_id(minion_id, cache_file):
         log.error('Could not cache minion ID: {0}'.format(exc))
 
 
+def eval_minion_id_func(opts):
+    '''
+    Evaluate minion ID function if 'minion_id_type' is 'func'
+    and return the result
+    '''
+    if '__minion_id_func_evaluated' in opts:
+        return opts['id']
+
+    import salt.loader as loader
```

Comment:
I don't believe we can move this import to the top. The reason is that `salt/loader.py` imports `salt.config` near the top of the file. If we add this import to the top of `salt/config/__init__.py` we would be creating a circular dependency which would break things. A comment explaining this is probably in order though.